sensu/goplugin.go

```diff
@@ -146,76 +169,56 @@ func (p *basePlugin) initPlugin() error {
 
 func (p *basePlugin) setupFlags(cmd *cobra.Command) error {
 	for _, opt := range p.options {
-		if err := setupFlag(cmd, opt); err != nil {
+		if err := opt.SetupFlag(cmd); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func setupFlag(cmd *cobra.Command, opt *PluginConfigOption) error {
-	if len(opt.Argument) == 0 {
+func (p *PluginConfigOption[T]) SetupFlag(cmd *cobra.Command) error {
```

Comment:
Looks like Default values may be have been dropped. Probably makes sense for them to get set here.

Reply:
Good call, I missed porting that case over. Surprised the tests didn't catch it, I'll add some to make sure it's covered.